sdk/nodejs/x/automation/stack.ts

```diff
@@ -174,16 +214,36 @@ export class Stack {
         }
 
         args.push("--exec-kind", kind);
+
+        let logPromise;
+        // Set up event log tailing
+        if (opts?.onEvent) {
+            const onEvent = opts.onEvent;
+            const logFile = createLogFile("up");
+            args.push("--event-log", logFile);
+
+            logPromise = this.readLines(logFile, (event) => {
+                onEvent(event);
+            });
+        }
+
+        const upPromise = this.runPulumiCmd(args, opts?.onOutput);
+        const promises: Promise<any>[] = logPromise ? [upPromise, logPromise] : [upPromise];
```

Comment:
Since `undefined` can be passed to `Promise.all` directly and just flows through as a no-op, could probably avoid this all - and just do `Promise.all([upPromise, logPromise])` below.

Related - but since you are kicking off both calls already ahead of this - it isn't strictly necessary to `Promise.all` at all - you could just `await` each sequentially.  Since they are already running concurrently, this will have the same effect.  But `Promise.all` is fine if you think it is clearer.

Reply:
I learned so many things in those 5 lines of text. I think I will keep the `Promise.all()` for clarity but I did actually think they didn't kick off until that line so 💡 🧠